Sources/XCTest/XCTestRun.swift

```diff
@@ -0,0 +1,150 @@
+    }
+
+    /// `true` if all tests in the run completed their execution without
+    /// recording any failures, otherwise `false`.
+    public var hasSucceeded: Bool {
+        guard stopped else {
+            return false
+        }
+        return totalFailureCount == 0
+    }
+
+    /// Designated initializer for the XCTestRun class.
+    /// - Parameter test: An XCTest instance.
+    /// - Returns: A test run for the provided test.
+    public required init(test: XCTest) {
+        self.test = test
+    }
+
+    /// Start a test run. Must not be called more than once.
+    public func start() {
+        guard !started else {
+            fatalError("Invalid attempt to start a test run that has " +
+                       "already been started: \(self)")
+        }
+        guard !stopped else {
+            fatalError("Invalid attempt to start a test run that has " +
+                       "already been stopped: \(self)")
+        }
+
+        startDate = NSDate()
+    }
+
+    /// Stop a test run. Must not be called unless the run has been started.
+    /// Must not be called more than once.
+    public func stop() {
+        guard started else {
+            fatalError("Invalid attempt to stop a test run that has " +
+                       "not yet been started: \(self)")
+        }
+        guard !stopped else {
+            fatalError("Invalid attempt to stop a test run that has " +
+                       "already been stopped: \(self)")
+        }
+
+        executionCount += 1
+        stopDate = NSDate()
+    }
+
+    /// Records a failure in the execution of the test for this test run. Must
+    /// not be called unless the run has been started. Must not be called if the
+    /// test run has been stopped.
+    /// - Parameter description: The description of the failure being reported.
+    /// - Parameter filePath: The file path to the source file where the failure
+    ///   being reported was encountered or nil if unknown.
+    /// - Parameter lineNumber: The line number in the source file at filePath
+    ///   where the failure being reported was encountered.
+    /// - Parameter expected: `true` if the failure being reported was the
+    ///   result of a failed assertion, `false` if it was the result of an
+    ///   uncaught exception.
+    public func recordFailureWithDescription(description: String, inFile filePath: String, atLine lineNumber: UInt, expected: Bool) {
+        guard started else {
+            fatalError("Invalid attempt to record a failure for a test run " +
+                       "that has not yet been started: \(self)")
+        }
+        guard !stopped else {
+            fatalError("Invalid attempt to record a failure for a test run " +
+                       "that has already been stopped: \(self)")
+        }
+
+        if expected {
+            failureCount += 1
+        } else {
+            unexpectedExceptionCount += 1
+        }
+    }
+
+    private var started: Bool {
```

Comment:
I think these would become `isStarted` and `isStopped` to follow the Swift 3 API guidelines.